pkg/pvc/pvc_protected_entity.go

```diff
@@ -326,6 +287,57 @@ func (this PVCProtectedEntity) getProtectedEntityForPV(ctx context.Context, pv *
 	return nil, errors.Errorf("Could not find PE for Persistent Volume %s", pv.Name)
 }
 
+func (this PVCProtectedEntity) getComponentsFromSnapshot(ctx context.Context, snapshotToDelete astrolabe.ProtectedEntitySnapshotID) ([]astrolabe.ProtectedEntity, error) {
+	this.logger.Infof("Attempting to get the components from the snapshot pe-id: %s", snapshotToDelete.String())
+	// PVC Component can be para-virt or ivd type.
+	snapshotID64Str := snapshotToDelete.String()
+	snapshotIDBytes, err := base64.RawStdEncoding.DecodeString(snapshotID64Str)
+	if err != nil {
+		errorMsg := fmt.Sprintf("Could not decode snapshot ID encoded string %s", snapshotID64Str)
+		this.logger.WithError(err).Error(errorMsg)
+		return []astrolabe.ProtectedEntity{}, errors.Wrap(err, errorMsg)
+	}
+	decodedPEID, err := astrolabe.NewProtectedEntityIDFromString(string(snapshotIDBytes))
+	if err != nil {
+		errorMsg := fmt.Sprintf("Could not translate decoded snapshotID %s into pe-id", string(snapshotIDBytes))
+		this.logger.WithError(err).Error(errorMsg)
+		return []astrolabe.ProtectedEntity{}, errors.Wrap(err, errorMsg)
+	}
+	// Determine the component pe-type, should be either ivd or para-virt
+	expectedComponentPEType := this.getComponentPEType()
+	if expectedComponentPEType == decodedPEID.GetPeType() {
+		// If the decoded snap is of the expected component pe-type return.
+		componentPE, err := this.ppetm.pem.GetProtectedEntity(ctx, decodedPEID)
+		if err != nil {
+			return []astrolabe.ProtectedEntity{}, errors.Wrapf(err, "Could not get Protected Entity for PE %s", decodedPEID.String())
+		}
+		this.logger.Infof("Retrieved the component PE %s from the snapshot pe-id: %s", componentPE.GetID().String(), snapshotToDelete.String())
+		return []astrolabe.ProtectedEntity{componentPE}, nil
+	}
+	if expectedComponentPEType == "ivd" && decodedPEID.GetPeType() == "paravirt-pv" {
```

Comment:
Do we already have "paravirt-pv" as constant somewhere that we can use here?

Reply:
Done.